[Doc] update queue.max_bytes and queue.page_capacity (#13833)
elastic/logstash: docs/static/persistent-queues.asciidoc

`queue.type`:: Specify `persisted` to enable persistent queues. By default, persistent queues are disabled (default: `queue.type: memory`).
`path.queue`:: The directory path where the data files will be stored. By default, the files are stored in `path.data/queue`. 

> `queue.page_capacity`:: The maximum size of a queue page in bytes. The queue data consists of append-only files called "pages". The default size is 64mb. Changing this value is unlikely to have performance benefits. You can change the page capacity of existing queue. The new size will apply to the new page.

Reviewer: This is confusing to me.  We say that "Changing this value is unlikely to have performance benefits," but then we give more info how to change it, so it seems like a mixed message. 

Are there _other_ benefits that we should call out?

Reviewer 2: One reason (likely not the only one):

Because each pipeline needs to have 1-2 pages open at any given time (the head that it is writing to, and the tail that it is reading from, which are often the same as each other), and doing so entails loading the page's file as memory-mapped, a logstash with many PQ-enabled pipelines (read:hundreds) can end up using a LOT of off-HEAP memory (64MB * 100 => 6GB). Reducing the page size can be a way to limit just how much memory is active per pipeline, which in turn allows a greater number of pipelines.

Author: we believe the page size, 64mb, is a good fit for most of the users, but at the same time, we do not stop them from tunning it to fit their situation. Here, I want to give a note that this value is changeable with an existing queue. Ry's example goes very deep about how PQ works inside. Probably, `queue.page_capacity` session is not the right place to share these details.

> `queue.page_capacity`:: The maximum size of a queue page in bytes. The queue data consists of append-only files called "pages". The default size is 64mb. Changing this value is unlikely to have performance benefits. You can change the page capacity of existing queue. The new size will apply to the new page.

Reviewer: ```suggestion
`queue.page_capacity`:: The queue data consists of append-only files called "pages." This value sets the maximum size of a queue page in bytes. 
The default size of 64mb is a good value for most users, and changing this value is unlikely to have performance benefits. 
If you change the page capacity of an existing queue, the new size applies only to the new page.
```

Reviewer: @kaisecheng, I tried to smooth the flow of the wording a bit. Let me know what you think.

The default is 1024mb (1gb).
+
Be sure that your disk has sufficient capacity to handle the cumulative total of `queue.max_bytes` across all persistent queues.
The total of `queue.max_bytes` for _all_ queues should be
lower than the capacity of your disk. 

> We do not check the disk has enough capacity for `queue.max_bytes` for all queues. The space check at startup only verify each queue has sufficient capacity to reach its `queue.max_bytes`.

Reviewer: ```suggestion
We do not check to see if the disk has enough capacity for `queue.max_bytes` for all queues. 
The space check at startup only verifies that each queue has sufficient capacity to reach its `queue.max_bytes`.
```

+
Be sure that your disk has sufficient capacity to handle the cumulative total of `queue.max_bytes` across all persistent queues.
The total of `queue.max.bytes` for _all_ queues should be

> lower than the capacity of your disk. The size is checked at startup.

Reviewer 2: IIRC, we check that _each_ queue has sufficient capacity to reach its max size, but we do not validate sufficient room for _all_ to reach max size. It may be worth rewording to indicate this.

Author: yes, we do not check space availability for the sum of max_bytes of all queues.
